### ISSUE TITLE
Same arguments result in duplicate HashCode in Component Configurations

### ISSUE THREAD

**Issue by yeldar-nurpeissov** [arkivanov/Decompose#733]
I have two components with config classes that have identical properties, like ID. Navigating from the first to the second component with the same hash-coded arguments causes an error: `java.lang.IllegalArgumentException: Key kotlin.Pair_173 was used multiple times.`

For example:
```
class SongDetailComponent {

    sealed interface Config {
        data class Detail(val id: String): Config
    }
}

class SingerDetailComponent {

    sealed interface Config {
        data class Detail(val id: String): Config
    }
}
```
If both song and singer have id = "1", navigating to the song screen works, but navigating to the singer screen results in the error.

The configs don't have to be in the same component to trigger this error.

IMHO, `hashString()` generates a hash from arguments.

![Screenshot 2024-07-08 074601](https://github.com/arkivanov/Decompose/assets/16415795/94530f94-f622-46b9-8b1f-60ece4c0272b)

`Children` Composable

![Screenshot 2024-07-08 074632](https://github.com/arkivanov/Decompose/assets/16415795/98335490-9294-43e5-8146-88e5e1ba5d5f)

Here is a [sample code](https://github.com/yeldar-nurpeissov/decompose-quick-guide/blob/test_fixed_duplicated_configs/composeApp/src/commonMain/kotlin/presentation/root/DefaultRootComponent.kt) that produces the error.

**Version**: [3.2.0-alpha01](https://github.com/arkivanov/Decompose/releases/tag/3.2.0-alpha01)

**Comment by arkivanov** [comment]
The `hashString` function also takes into account `qualifiedName` of the configuration class. This is a bug introduced in the recent version `3.2.0-alpha01` with `DecomposeExperimentFlags.duplicateConfigurationsEnabled` flag enabled. I will fix it, thanks!

**Comment by arkivanov** [comment]
Should be fixed in version [3.2.0-alpha02](https://github.com/arkivanov/Decompose/releases/tag/3.2.0-alpha02).
